tests/unit/development/test_base_component.py

```diff
@@ -55,71 +57,67 @@ def nested_tree():
     return c, c1, c2, c3, c4, c5
 
 
-class TestComponent(unittest.TestCase):
+class TestComponent:
     def test_init(self):
         Component(a=3)
 
     def test_get_item_with_children(self):
         c1 = Component(id='1')
         c2 = Component(children=[c1])
-        self.assertEqual(c2['1'], c1)
+        assert c2['1'] == c1
 
     def test_get_item_with_children_as_component_instead_of_list(self):
         c1 = Component(id='1')
         c2 = Component(id='2', children=c1)
-        self.assertEqual(c2['1'], c1)
+        assert c2['1'] == c1
 
     def test_get_item_with_nested_children_one_branch(self):
         c1 = Component(id='1')
         c2 = Component(id='2', children=[c1])
         c3 = Component(children=[c2])
-        self.assertEqual(c2['1'], c1)
-        self.assertEqual(c3['2'], c2)
-        self.assertEqual(c3['1'], c1)
+        assert c2['1'] == c1
+        assert c3['2'] == c2
+        assert c3['1'] == c1
 
     def test_get_item_with_nested_children_two_branches(self):
         c1 = Component(id='1')
         c2 = Component(id='2', children=[c1])
         c3 = Component(id='3')
         c4 = Component(id='4', children=[c3])
         c5 = Component(children=[c2, c4])
-        self.assertEqual(c2['1'], c1)
-        self.assertEqual(c4['3'], c3)
-        self.assertEqual(c5['2'], c2)
-        self.assertEqual(c5['4'], c4)
-        self.assertEqual(c5['1'], c1)
-        self.assertEqual(c5['3'], c3)
+        assert c2['1'] == c1
+        assert c4['3'] == c3
+        assert c5['2'] == c2
+        assert c5['4'] == c4
+        assert c5['1'] == c1
+        assert c5['3'] == c3
 
     def test_get_item_with_nested_children_with_mixed_strings_and_without_lists(self):  # noqa: E501
         c, c1, c2, c3, c4, c5 = nested_tree()
         keys = [k for k in c]
-        self.assertEqual(
-            keys,
-            [
-                '0.0',
-                '0.1',
-                '0.1.x',
-                '0.1.x.x',
-                '0.1.x.x.0'
-            ]
-        )
+
+        assert keys == [
+            '0.0',
+            '0.1',
+            '0.1.x',
+            '0.1.x.x',
+            '0.1.x.x.0'
+        ]
 
         # Try to get each item
         for comp in [c1, c2, c3, c4, c5]:
-            self.assertEqual(c[comp.id], comp)
+            assert c[comp.id] == comp
 
         # Get an item that doesn't exist
-        with self.assertRaises(KeyError):
+        with pytest.raises(KeyError):
             c['x']
 
     def test_len_with_nested_children_with_mixed_strings_and_without_lists(self):  # noqa: E501
         c = nested_tree()[0]
-        self.assertEqual(
-            len(c),
-            5 +  # 5 components
-            5 +  # c2 has 2 strings, 2 numbers, and a None
-            1    # c1 has 1 string
-        )
+        # 5 components
+        # c2 has 2 strings, 2 numbers, and a None
+        # c1 has 1 string
+        assert len(c) == 5 + 5 + 1
```

Comment:
:cow: minor: a translation pattern we start doing now is to also translate the comment into an assertion message, by using clearly SHOULD or SHOULD NOT, so here we can do sth like

`assert len(c) == 5 + 5 + 1, "the length of the nested children should match the total of 5 components, 2 strings + 2 numbers + none in c2, and 1 string in c1"

another practice is to use `black` as an automatic formatting tool, it's inspired by the awesome `golang` official formatter.  we use `fmt: off` and `fmt: on` to disagree with the default style, but unlike in `yapf` world, people argument endlessly about the options and settings.

Reply:
+1 for black. As a matter of fact, I intentionnally turned it off because I though it would not play well with flake8 in CI, but used it anyway on `test_configs`